File: app/src/main/java/com/mvc/cryptovault_android/view/DialogHelper.java
package com.mvc.cryptovault_android.view;

import android.app.Dialog;
import android.content.Context;
import android.os.Handler;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.mvc.cryptovault_android.R;
import com.mvc.cryptovault_android.listener.IDialogViewClickListener;

public class DialogHelper {
    private Dialog mDialog;
    private static DialogHelper mDialogHelper;

    public static DialogHelper getInstance() {
        if (mDialogHelper == null) {
            synchronized (DialogHelper.class) {
                mDialogHelper = new DialogHelper();
            }
        }
        return mDialogHelper;
    }

    public Dialog create(Context context, int resId) {
        mDialog = new Dialog(context, R.style.tras_dialog);
        mDialog.setCanceledOnTouchOutside(false);
        mDialog.setCancelable(false);
        View dialogView = LayoutInflater.from(context).inflate(resId, null);
        mDialog.setContentView(dialogView);
        return mDialog;
    }

    public Dialog create(Context context, int resId, String msg) {
        mDialog = new Dialog(context, R.style.tras_dialog);
        mDialog.setCanceledOnTouchOutside(false);
        mDialog.setCancelable(false);
        View dialogView = LayoutInflater.from(context).inflate(R.layout.layout_dialog, null);
        ImageView icon = dialogView.findViewById(R.id.dialog_icon);
        TextView title = dialogView.findViewById(R.id.dialog_title);
        Glide.with(context).load(resId).into(icon);
        title.setText(msg);
        mDialog.setContentView(dialogView);
        return mDialog;
    }

    public Dialog create(Context context, String msg, IDialogViewClickListener clickListener) {
        mDialog = new Dialog(context, R.style.tras_dialog);
        mDialog.setCanceledOnTouchOutside(false);
        mDialog.setCancelable(false);
        View dialogView = LayoutInflater.from(context).inflate(R.layout.layout_sub_dialog, null);
        TextView cancle = dialogView.findViewById(R.id.hint_cancle);
        TextView enter = dialogView.findViewById(R.id.hint_enter);
        TextView title = dialogView.findViewById(R.id.hint_title);
        title.setText(msg);
        mDialog.setContentView(dialogView);
        cancle.setOnClickListener(v -> clickListener.click(cancle.getId()));
        enter.setOnClickListener(v -> clickListener.click(enter.getId()));
        return mDialog;
    }

    public void resetDialogResource(Context context, int resId, String msg) {
        View dialogView = LayoutInflater.from(context).inflate(R.layout.layout_dialog, null);
        ImageView icon = dialogView.findViewById(R.id.dialog_icon);
        TextView title = dialogView.findViewById(R.id.dialog_title);
        Glide.with(context).load(resId).into(icon);
        title.setText(msg);
        mDialog.setContentView(dialogView);
    }

    public void dismissDelayed(IDialogDialog dialogDialog) {
        new Handler().postDelayed(() -> {
            dismiss();
            if (dialogDialog != null) {
                dialogDialog.callback();
            }
        }, 2000);
    }

    public void dismissDelayed(IDialogDialog dialogDialog, long delayMillis) {
        new Handler().postDelayed(() -> {
            dismiss();
            if (dialogDialog != null) {
                dialogDialog.callback();
            }
        }, delayMillis);
    }

    public void dismiss() {
        if (mDialog != null) {
            mDialog.dismiss();
        }
    }

    public interface IDialogDialog {
        void callback();
    }
}

File: app/src/main/java/com/mvc/cryptovault_android/view/DialogHelper.kt
package com.mvc.cryptovault_android.view

import android.app.Dialog
import android.content.Context
import android.os.Handler
import android.view.LayoutInflater
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide
import com.mvc.cryptovault_android.R
import com.mvc.cryptovault_android.listener.IDialogViewClickListener

class DialogHelper {
    private lateinit var mDialog: Dialog

    fun create(context: Context, resId: Int): Dialog {
        mDialog = Dialog(context, R.style.tras_dialog)
        mDialog.setCanceledOnTouchOutside(false)
        mDialog.setCancelable(false)
        val dialogView = LayoutInflater.from(context).inflate(resId, null)
        mDialog.setContentView(dialogView)
        return mDialog
    }

    fun create(context: Context, resId: Int, msg: String): Dialog {
        mDialog = Dialog(context, R.style.tras_dialog)
        mDialog.setCanceledOnTouchOutside(false)
        mDialog.setCancelable(false)
        val dialogView = LayoutInflater.from(context).inflate(R.layout.layout_dialog, null)
        val icon = dialogView.findViewById<ImageView>(R.id.dialog_icon)
        val title = dialogView.findViewById<TextView>(R.id.dialog_title)
        Glide.with(context).load(resId).into(icon)
        title.text = msg
        mDialog.setContentView(dialogView)
        return mDialog
    }

    fun create(context: Context, clickListener: IDialogViewClickListener,msg:String): Dialog {
        mDialog = Dialog(context, R.style.tras_dialog)
        mDialog.setCanceledOnTouchOutside(false)
        mDialog.setCancelable(false)
        val dialogView = LayoutInflater.from(context).inflate(R.layout.layout_skip_dialog, null)
        dialogView.findViewById<TextView>(R.id.dialog_title).text = msg
        val submit = dialogView.findViewById<TextView>(R.id.dialog_submit)
        submit.setOnClickListener { clickListener.click(submit.id) }
        mDialog.setContentView(dialogView)
        return mDialog
    }

    fun create(context: Context, msg: String, clickListener: IDialogViewClickListener): Dialog {
        mDialog = Dialog(context, R.style.tras_dialog)
        mDialog.setCanceledOnTouchOutside(false)
        mDialog.setCancelable(false)
        val dialogView = LayoutInflater.from(context).inflate(R.layout.layout_sub_dialog, null)
        val cancle = dialogView.findViewById<TextView>(R.id.hint_cancle)
        val enter = dialogView.findViewById<TextView>(R.id.hint_enter)
        val title = dialogView.findViewById<TextView>(R.id.hint_title)
        title.text = msg
        mDialog.setContentView(dialogView)
        cancle.setOnClickListener { v -> clickListener.click(cancle.id) }
        enter.setOnClickListener { v -> clickListener.click(enter.id) }
        return mDialog
    }

    fun resetDialogResource(context: Context, resId: Int, msg: String) {
        val dialogView = LayoutInflater.from(context).inflate(R.layout.layout_dialog, null)
        val icon = dialogView.findViewById<ImageView>(R.id.dialog_icon)
        val title = dialogView.findViewById<TextView>(R.id.dialog_title)
        Glide.with(context).load(resId).into(icon)
        title.text = msg
        mDialog.setContentView(dialogView)
    }

    fun dismissDelayed(dialogDialog: IDialogDialog?) {
        Handler().postDelayed({
            dismiss()
            dialogDialog?.callback()
        }, 2000)
    }

    fun dismissDelayed(dialogDialog: IDialogDialog?, delayMillis: Long) {
        Handler().postDelayed({
            dismiss()
            dialogDialog?.callback()
        }, delayMillis)
    }

    fun dismiss() {
        if (mDialog != null) {
            mDialog.dismiss()
        }
    }

    interface IDialogDialog {
        fun callback()
    }

    companion object {
        val instance: DialogHelper by lazy(mode = LazyThreadSafetyMode.SYNCHRONIZED) {
            DialogHelper()
        }
    }
}
